autopilot/contractor/hostscore.go

```diff
@@ -21,8 +21,29 @@ const (
 	// minValidScore is the smallest score that a host can have before
 	// being ignored.
 	minValidScore = math.SmallestNonzeroFloat64
+
+	// minSubScore is the smallest score that a host can have for a given
+	// sub-score such as its pricing. By choosing 0.1, a host that has a very bad
+	// price score but is otherwise perfect can at most be 90% less likely to be
+	// picked than a host that has a perfect score.
+	minSubScore = 0.1
 )
 
+// clampScore makes sure that a score can not be smaller than 'minSubScore'.
+// Unless the score is 0, which indicates a more severe issue with the host. By
+// doing so, we limit the impact of a single sub-score on the overall score of a
+// host.
+func clampScore(score float64) float64 {
+	if score <= 0.0 {
```

Comment:
We should pretty much have that. Because we shouldn't be handing out a flat 0.0 unless your protocol version is below 1.5.9. Which means you still run siad.